app/scripts/modules/kubernetes/src/v2/pipelines/stages/deployManifest/PreRewriteManifestBindArtifactSelector.tsx

```diff
@@ -0,0 +1,70 @@
+import * as React from 'react';
+import Select, { Option } from 'react-select';
+import { find, map } from 'lodash';
+
+import { ArtifactIcon, ExpectedArtifactService, IExpectedArtifact, IPipeline, IStage } from '@spinnaker/core';
+
+import { IManifestBindArtifact } from './ManifestBindArtifactsSelector';
+
+export interface IExpectedArtifactMultiSelectorProps {
+  pipeline: IPipeline;
+  stage: IStage;
+  bindings: IManifestBindArtifact[];
+  onChangeBindings: (bindings: IManifestBindArtifact[]) => void;
+}
+
+export class PreRewriteManifestBindArtifactSelector extends React.Component<IExpectedArtifactMultiSelectorProps> {
+  private getAvailableExpectedArtifacts = (): IExpectedArtifact[] => {
+    return ExpectedArtifactService.getExpectedArtifactsAvailableToStage(this.props.stage, this.props.pipeline);
+  };
+
+  private getOptions = (): Array<Option<string>> => {
+    return map(this.getAvailableExpectedArtifacts(), a => ({
```

Comment:
nit: curious as to why lodash `map` over native here

Reply:
no good reason, just prettier :nail_care:

Reply:
(and null-safe)